src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookExplorerViewlet.ts

```diff
@@ -261,17 +261,30 @@ export class NotebookExplorerViewPaneContainer extends ViewPaneContainer {
 		}
 
 		this.validateQuery(query).then(() => {
+			let filesToIncludeFiltered: string = '';
 			if (this.views.length > 1) {
-				let filesToIncludeFiltered: string = '';
 				this.views.forEach(async (v) => {
 					const { treeView } = (<ITreeViewDescriptor>Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).getView(v.id));
-					let items = await treeView?.dataProvider.getChildren();
-					items?.forEach(root => {
+					let root = treeView?.root;
+					let items = await treeView?.dataProvider.getChildren(root);
+					items?.forEach(async root => {
+						let contentFolder: URI;
 						this.updateViewletsState();
-						let folderToSearch: IFolderQuery = { folder: URI.file(path.join(isString(root.tooltip) ? root.tooltip : root.tooltip.value, 'content')) };
-						query.folderQueries.push(folderToSearch);
-						filesToIncludeFiltered = filesToIncludeFiltered + path.join(folderToSearch.folder.fsPath, '**', '*.md') + ',' + path.join(folderToSearch.folder.fsPath, '**', '*.ipynb') + ',';
-						this.searchView.startSearch(query, null, filesToIncludeFiltered, false, this.searchWidget);
+						let rootFolder = isString(root.tooltip) ? root.tooltip : root.tooltip.value;
+						let folderToSearch: IFolderQuery;
+						if (root.contextValue !== 'pinnedNotebook') {
+							contentFolder = URI.file(rootFolder);
+							(await treeView?.dataProvider.getChildren(root)).map((child) => {
```

Comment:
Can you point to where this broke? getChildren says it accepts undefined and in all my workings with it undefined has always meant "get all the children of the root" - changing this would be a HUGE breaking change from what I understand so I'm very surprised that this would be broken in this way.

Reply:
When I don't passed any arguments it returns undefined. I noticed that the method triggers a call to the method `postGetChildren` but it doesn't await the results. 
I'm not sure how wait for the postGetChildren results.
https://github.com/microsoft/azuredatastudio/blob/main/src/vs/workbench/api/browser/mainThreadTreeViews.ts#L181